edb/edgeql/compiler/viewgen.py

```diff
@@ -394,6 +339,107 @@ def _process_view(
+        if not default_expr:
+            if ptrcls.get_required(ctx.env.schema) and pn != sn.UnqualName(
+                "__type__"
+            ):
+                if ptrcls.is_property(ctx.env.schema):
+                    # If the target is a sequence, there's no need
+                    # for an explicit value.
+                    ptrcls_target = ptrcls.get_target(ctx.env.schema)
+                    assert ptrcls_target is not None
+                    if ptrcls_target.issubclass(
+                        ctx.env.schema,
+                        ctx.env.schema.get(
+                            "std::sequence", type=s_objects.SubclassableObject
+                        ),
+                    ):
+                        continue
+                vn = ptrcls.get_verbosename(ctx.env.schema, with_parent=True)
+                raise errors.MissingRequiredError(
+                    f"missing value for required {vn}"
+                )
+            else:
+                continue
+
+        ptrcls_sn = ptrcls.get_shortname(ctx.env.schema)
+        default_ql = qlast.ShapeElement(
+            expr=qlast.Path(
+                steps=[
+                    qlast.Ptr(
+                        ptr=qlast.ObjectRef(
+                            name=ptrcls_sn.name,
+                            module=ptrcls_sn.module,
+                        ),
+                    ),
+                ],
+            ),
+            compexpr=qlast.DetachedExpr(
+                expr=default_expr.qlast,
+                preserve_path_prefix=True,
+            ),
+        )
+
+        with ctx.new() as scopectx:
+            result.append(
+                _normalize_view_ptr_expr(
+                    ir_set,
+                    default_ql,
+                    view_scls,
+                    path_id=path_id,
+                    path_id_namespace=path_id_namespace,
+                    exprtype=exprtype,
+                    from_default=True,
+                    view_rptr=view_rptr,
+                    ctx=scopectx,
+                ),
+            )
+
+    schema = ctx.env.schema
+
+    # toposort defaults 
+    pointer_indexes = {}
+    for (index, (pointer, _)) in enumerate(result):
+        name = pointer.get_bases(schema).first(schema).get_name(schema).name
+        pointer_indexes[name] = index
+    graph = {}
+    for (index, (_, irset)) in enumerate(result):
+        assert irset
+        dep_pointers = irutils.collect_pointers(irset)
+        dep_names = (
+            pointer.target.path_id.rptr_name() for pointer in dep_pointers
+            if pointer.source.typeref.id == stype.id
+        )
+        deps = {
+            pointer_indexes[name.name] for name in dep_names 
+            if name and name.name in pointer_indexes
+        }
+        graph[index] = topological.DepGraphEntry(
+            item=index, deps=deps, extra=False,
+        )
+
+    return [result[i] for i in topological.sort(graph, allow_unresolved=True)]
```

Comment:
This is the new part: toposort of pointers.

It does not work though, I don't know how to get id or name of the base property from the derived property.

Reply:
`get_nearest_non_derived_parent` might be what you want

We need some substantial naming reform.